synapse/lib/layer.py

```diff
@@ -1204,6 +1224,99 @@ async def __anit__(self, layrinfo, dirn, nexsroot=None, allow_upstream=True, map
 
         self.onfini(self._onLayrFini)
 
+        # if we are a mirror, we upstream all our edits and
+        # wait for them to make it back down the pipe...
+        self.leader = None
+        self.leadtask = None
+        self.ismirror = layrinfo.get('mirror') is not None
+
+    @contextlib.contextmanager
+    def getIdenFutu(self, iden=None):
+
+        if iden is None:
+            iden = s_common.guid()
+
+        futu = self.loop.create_future()
+        self.futures[iden] = futu
+
+        yield iden, futu
+
+        self.futures.pop(iden, None)
+
+    async def getMirrorStatus(self):
+        # TODO plumb back to upstream on not self.core.isactive
+        retn = {'mirror': self.leader is not None}
+        if self.leader:
+            proxy = await self.leader.proxy()
+            retn['local'] = {'size': await self.getEditSize()}
+            retn['remote'] = {'size': await proxy.getEditSize()}
+        return retn
+
+    async def initLayerActive(self):
+
+        if self.leadtask is not None:
+            self.leadtask.cancel()
+
+        mirror = self.layrinfo.get('mirror')
+        if mirror is not None:
+            self.leader = await s_telepath.Client.anit(mirror)
```

Comment:
Can we set `conf={'retrysleep': 1}` to tone down logspam when trying to connect in the event of a downstream that isn't online?